torch_frame/data/dataset.py

```diff
@@ -303,10 +338,16 @@ def __init__(
 
         self.text_embedder_cfg = text_embedder_cfg
         self.text_tokenizer_cfg = text_tokenizer_cfg
-        self.col_to_sep = canonicalize_col_to_sep(col_to_sep, [
+        self.col_to_sep = canonicalize_col_to_pattern(col_to_sep, [
             col for col, stype in self.col_to_stype.items()
             if stype == torch_frame.multicategorical
         ])
+        self.col_to_time_format = (
+            None if col_to_time_format is None else
+            canonicalize_col_to_pattern(col_to_time_format, [
+                col for col, stype in self.col_to_stype.items()
+                if stype == torch_frame.timestamp
+            ]))
```

Comment:
Same here. Let's canonicalize `None` as well. The output here should always be a dictionary mapping from col name into time_format (which can be `None`).